### DISCUSSION TITLE
What do we mean by "experimental"?

### DISCUSSION THREAD

**Discussion by craigmac** [neovim/neovim#25745] [General]
As the title says, what do we mean by 'experimental' in the docs? While doing `:helpgrep experimental \| copen` and browsing through some of the results, the messaging isn't clear on the definition of experimental; does it mean that it may be completely removed at any time from Neovim, or is it here to stay? And at what point is the experimental label removed? I guess what I'm really after an understanding of the graduation process from experimental to just another feature.

**Comment by clason** [comment]
Can you be more explicit? In general, "experimental" means we reserve the right to make any changes we see fit, including breaking changes without a formal deprecation cycle: "Use at your own risk, or wait until we are happy with it and declare it stable."

Conceivably, this _could_ include removing it altogether ("no promises" means no promises); in practice that is vanishingly unlikely after we've already spent several personyears on it.

**Reply by craigmac** [reply]
Ok, let's say I want to use the  `set cmdheight=0` feature - it's marked as experimental currently in 0.9.4, so I want to wait until bugs/etc. are all fixed for it, does that mean that in some future release there'll be a news.txt item saying `cmdheight=0` in now considered stable?

**Reply by clason** [reply]
Maybe; or the experimental label is just silently dropped. IIRC, we have yet to successfully graduate an experimental feature, so there's no precedent to go by, and we'll have burn that bridge when we cross it for the first time.
